packages/cli-repl/src/smoke-tests.ts

```diff
@@ -0,0 +1,33 @@
+import { spawn } from 'child_process';
+import assert from 'assert';
+import { once } from 'events';
+
+// Run smoke tests on a executable, e.g. runSmokeTests("/path/to/mongosh.exe")
+// or runSmokeTests("/path/to/node", "packages/cli-repl/bin/mongosh.js").
+export async function runSmokeTests(executable: string, ...args: string[]): Promise<void> {
+  for (const { input, output, testArgs } of [{
+    input: 'print("He" + "llo" + " Wor" + "ld!")',
+    output: /Hello World!/,
+    testArgs: ['--nodb']
+  }]) {
+    await runSmokeTest(executable, [...args, ...testArgs], input, output);
+  }
+  // eslint-disable-next-line no-console
+  console.log('all tests passed');
+}
+
+async function runSmokeTest(executable: string, args: string[], input: string, output: RegExp): Promise<void> {
+  const proc = spawn(executable, [...args], {
```

Comment:
Won't this suffer the same issue on Windows that we had e.g. with lerna where we would have to rely on `cross-spawn` to get it working? I'd assume on a Windows machine the executable (when self-tested) is `mongosh.exe` and then `spawn` will fail?

Reply:
Err, we can see what CI says, but I’m not sure why this would fail? I think `cross-spawn` mostly is for situations in which we don’t have the full path to what we want to run and want to emulate `cmd`’s lookup algorithm for things like `PATH_EXT`, but that isn’t really a concern here

Reply:
Ah okay, thank you :)